circe/src/main/scala/org/http4s/circe/middleware/JsonDebugErrorHandler.scala

```diff
@@ -18,31 +18,31 @@ package org.http4s.circe.middleware
 
 import cats.data._
 import cats.effect._
+import cats.syntax.all._
 import io.circe._
 import io.circe.syntax._
 import org.http4s._
 import org.http4s.circe._
 import org.http4s.headers.Connection
 import org.typelevel.ci._
+import org.typelevel.log4cats.LoggerFactory
 
 object JsonDebugErrorHandler {
-  private[this] val messageFailureLogger =
-    Platform.loggerFactory.getLoggerFromName(
+  private[this] def messageFailureLogger[F[_]: LoggerFactory] =
+    LoggerFactory[F].getLoggerFromName(
       "org.http4s.circe.middleware.jsondebugerrorhandler.message-failures"
     )
-  private[this] val serviceErrorLogger =
-    Platform.loggerFactory.getLoggerFromName(
+  private[this] def serviceErrorLogger[F[_]: LoggerFactory] =
+    LoggerFactory[F].getLoggerFromName(
       "org.http4s.circe.middleware.jsondebugerrorhandler.service-errors"
     )
 
   // Can be parametric on my other PR is merged.
-  def apply[F[_]: Concurrent, G[_]](
+  def apply[F[_]: Concurrent: LoggerFactory, G[_]](
       service: Kleisli[F, Request[G], Response[G]],
       redactWhen: CIString => Boolean = Headers.SensitiveHeaders.contains,
   ): Kleisli[F, Request[G], Response[G]] =
     Kleisli { req =>
-      import cats.syntax.applicative._
-      import cats.syntax.applicativeError._
       implicit def entEnc[M[_]]: EntityEncoder.Pure[JsonErrorHandlerResponse[M]] =
```

Comment:
Not related to the PR, but it caught my eye.
 The `entEnc` is always used with `G` type parameter. Technically, we can preallocate it too:

```scala
implicit val entityEncoder: EntityEncider.Pure[JsonErrorHandlerResponse[F]] = 
  JsonErrorHandlerResponse.entEnc[M](redactWhen)
```